test/test_linearizer_failures.py

```diff
@@ -236,6 +236,8 @@ def test_failure_31(self):
     opts = [Opt(op=OptOps.UNROLL, axis=0, amt=0), Opt(op=OptOps.PADTO, axis=1, amt=32)]
     helper_test_lin(Linearizer(ast), opts=opts, failed_platforms=[])
 
+  # AssertionError: failed on METAL with COMPARE_ERROR
+  @unittest.skipIf(Device.DEFAULT=="METAL", "flaky on METAL")
```

Comment:
This only fails on local Metal, not CI Metal?

Reply:
yes, only local for me atleast.

Reply:
I can't repro, Metal 3 chip M2 max

Reply:
not sure metal version, but m3 pro
if its only for me, then its probably ok to close
freshly cloned repo 898430c004df1fa97a1c613ffb356a09b8175abe

```
$ python3 -m pytest test/test_linearizer_failures.py
========================================================== test session starts ===========================================================
platform darwin -- Python 3.8.19, pytest-8.2.2, pluggy-1.5.0
rootdir: /Users/willygorillatg/tinygrad
plugins: hypothesis-6.103.1, xdist-3.6.1
collected 31 items                                                                                                                       

test/test_linearizer_failures.py ..............s..........F.....                                                                   [100%]

================================================================ FAILURES ================================================================
_________________________________________________ TestLinearizerFailures.test_failure_32 _________________________________________________

self = <test.test_linearizer_failures.TestLinearizerFailures testMethod=test_failure_32>

    @unittest.skipIf(CI, "for real AMD GPU")
    def test_failure_32(self):
      # kernel from beaming resnet
      # Memory access fault on tinybox red
      ast = LazyOp(op=BufferOps.STORE, src=(LazyOp(op=UnaryOps.CAST, src=(LazyOp(op=ReduceOps.SUM, src=(LazyOp(op=UnaryOps.CAST, src=(LazyOp(op=BinaryOps.MUL, src=(LazyOp(op=BufferOps.LOAD, src=(), arg=MemBuffer(idx=1, dtype=dtypes.half, st=ShapeTracker(views=(View(shape=(1, 256, 1, 256, 4, 16, 4, 16), strides=(0, 50176, 0, 196, 0, 14, 0, 1), offset=-15, mask=((0, 1), (0, 256), (0, 1), (0, 256), (0, 4), (1, 15), (0, 4), (1, 15)), contiguous=False), View(shape=(256, 1, 256, 14, 14, 256, 3, 3), strides=(1048576, 0, 0, 64, 1, 4096, 1088, 17), offset=0, mask=None, contiguous=False))))), LazyOp(op=BufferOps.LOAD, src=(), arg=MemBuffer(idx=2, dtype=dtypes.half, st=ShapeTracker(views=(View(shape=(256, 1, 256, 14, 14, 256, 3, 3), strides=(0, 0, 2304, 0, 0, 9, 3, 1), offset=0, mask=None, contiguous=False),))))), arg=None),), arg=dtypes.float),), arg=(7, 6, 5)),), arg=dtypes.half),), arg=MemBuffer(idx=0, dtype=dtypes.half, st=ShapeTracker(views=(View(shape=(256, 1, 256, 14, 14, 1, 1, 1), strides=(50176, 0, 196, 14, 1, 0, 0, 0), offset=0, mask=None, contiguous=True),))))
      opts = [Opt(op=OptOps.TC, axis=2, amt=2), Opt(op=OptOps.UPCAST, axis=2, amt=7), Opt(op=OptOps.UNROLL, axis=1, amt=0), Opt(op=OptOps.LOCAL, axis=1, amt=16)]
>     helper_test_lin(Linearizer(ast), opts=opts, failed_platforms=[])

test/test_linearizer_failures.py:245: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

lin = <tinygrad.codegen.linearizer.Linearizer object at 0x137fecd30>
opts = [Opt(op=OptOps.TC, axis=2, amt=2), Opt(op=OptOps.UPCAST, axis=2, amt=7), Opt(op=OptOps.UNROLL, axis=1, amt=0), Opt(op=OptOps.LOCAL, axis=1, amt=16)]
failed_platforms = [], rtol = 0.01, atol = 0.01

    def helper_test_lin(lin: Linearizer, opts, failed_platforms, rtol=1e-2, atol=1e-2):
      if any(b.dtype == dtypes.half for b in lin.membufs) and not is_dtype_supported(dtypes.half): return
      if any(b.dtype == dtypes.bfloat16 for b in lin.membufs) and not is_dtype_supported(dtypes.bfloat16): return
    
      for opt in opts:
        try:
          lin.apply_opt(opt)
        except KernelOptError:
          # it's considered fixed if we invalidated the opts
          assert Device.DEFAULT not in failed_platforms, f"unexpected success on {Device.DEFAULT}"
          return
    
      compare_result = compare_linearizer(lin, rtol=rtol, atol=atol)
      if compare_result[0] in ["PASS", "KernelOptError"]:
        # it's considered fixed if we invalidated the opts
        assert Device.DEFAULT not in failed_platforms, f"unexpected success on {Device.DEFAULT}"
      else:
>       assert Device.DEFAULT in failed_platforms, f"failed on {Device.DEFAULT} with {compare_result[0]}"
E       AssertionError: failed on METAL with COMPARE_ERROR
E       assert 'METAL' in []
E        +  where 'METAL' = <tinygrad.device._Device object at 0x1070eab80>.DEFAULT

test/test_linearizer_failures.py:33: AssertionError
======================================================== short test summary info =========================================================
FAILED test/test_linearizer_failures.py::TestLinearizerFailures::test_failure_32 - AssertionError: failed on METAL with COMPARE_ERROR
================================================ 1 failed, 29 passed, 1 skipped in 7.66s =================================================
```